src/utils/withData.js

```diff
@@ -0,0 +1,91 @@
+/* istanbul ignore file */
+// TODO create unit tests for this file
+import { connect } from 'react-redux';
+import React from 'react';
+import { getAPIClient } from './api/network';
+
+function withData(apis = {}) {
+  return function (WrappedComponent) {
+    class DataProvider extends React.Component {
+      constructor() {
+        super();
+
+        this.state = {
+          apis: Object.keys(apis).reduce((acc, key) => {
+            acc[key] = {
+              data: [],
+              error: '',
+              isLoading: false,
+            };
+            return acc;
+          }, {}),
+        };
+
+        this.loadData = this.loadData.bind(this);
+        this.setApiState = this.setApiState.bind(this);
+      }
+
+      componentDidMount() {
+        Object.keys(apis).forEach((key) => {
+          if (apis[key].autoload) {
+            this.loadData(key);
+          }
+        });
+      }
+
+      setApiState(key, newState) {
+        this.setState({
+          apis: {
+            ...this.state.apis,
+            [key]: {
+              ...this.state.api[key],
+              ...newState,
+            },
+          },
+        });
+      }
+
+      loadData(key, params = {}, ...args) {
+        const { apiClient, apiParams } = this.props;
+        this.setApiState(key, { isLoading: true });
+        apis[key].apiUtil(apiClient, { ...apiParams[key], ...params }, ...args).then((data) => {
+          this.setApiState(key, { data, isLoading: false });
+        }).catch((error) => {
+          this.setApiState(key, { error, isLoading: false });
+        });
+      }
+
+      render() {
+        const { apiClient, apiParams, ...restOfProps } = this.props;
+        return (
+          <WrappedComponent
+            {...{
+              ...Object.keys(this.state.apis).reduce((acc, key) => {
```

Comment:
Since it uses the `Object.keys` multiple times, isn't  it possible to set this into some variable on the constructor so there is no  need to fetch the keys from the Object again?

Reply:
Fixed in https://github.com/LiskHQ/lisk-hub/pull/2258/commits/e383e75411c439b01f35abcd7d6cea6d804d2aae